### PR TITLE
Sonar fix blocker and some critical

### PR DESCRIPTION
**Pull request by isvonja**

### CODE REVIEW

File: whois-api/src/main/java/net/ripe/db/whois/api/rest/GeolocationService.java [hunk 1]
```diff
@@ -116,7 +119,7 @@ public WhoisResources geolocation(
                     }
                 }
             } catch (EmptyResultDataAccessException ignored) {
-                // stop looking
+                LOGGER.debug("EmptyResultDataAccessException ", ignored.getMessage());
```
Comment (eshryane) [inline whois-api/src/main/java/net/ripe/db/whois/api/rest/GeolocationService.java]: the message doesn't get logged, is this intentional?

File: whois-api/src/main/java/net/ripe/db/whois/api/rest/ReferencesService.java [hunk 2]
```diff
@@ -270,16 +270,16 @@ public WhoisResources performUpdates(
...
-                    // continue
-                } else if (status == UpdateStatus.FAILED_AUTHENTICATION) {
-                    throw new ReferenceUpdateFailedException(Response.Status.UNAUTHORIZED, whoisResources);
-                } else if (status == UpdateStatus.EXCEPTION) {
-                    throw new ReferenceUpdateFailedException(Response.Status.INTERNAL_SERVER_ERROR, whoisResources);
-                } else if (updateContext.getMessages(update).contains(UpdateMessages.newKeywordAndObjectExists())) {
-                    throw new ReferenceUpdateFailedException(Response.Status.CONFLICT, whoisResources);
-                } else {
-                    throw new ReferenceUpdateFailedException(Response.Status.BAD_REQUEST, whoisResources);
+                if (status != UpdateStatus.SUCCESS) {
```
Comment (eshryane) [inline whois-api/src/main/java/net/ripe/db/whois/api/rest/ReferencesService.java]: much more readable

File: whois-api/src/main/java/net/ripe/db/whois/api/transfer/logic/inetnum/InetnumTransfer.java [hunk 3]
```diff
@@ -31,7 +31,7 @@ public static boolean isIanaResource(final RpslObject resource) {
         final RpslAttribute netname = resource.findAttribute(AttributeType.NETNAME);
 
         final boolean status = (netname != null && IANA_NETNAME.equals(netname.getValue().trim()));
-        LOGGER.debug("Is inetnum {} with netname {} iana? {}", resource.getKey(), netname.getValue(), status);
+        LOGGER.debug("Is inetnum {} with netname {} iana? {}", resource.getKey(), netname != null ? netname.getValue() : null, status);
```
Comment (eshryane) [inline whois-api/src/main/java/net/ripe/db/whois/api/transfer/logic/inetnum/InetnumTransfer.java]: this appears to be a useless check - null is still used

File: whois-commons/src/main/java/net/ripe/db/whois/common/MaintenanceMode.java [hunk 4]
```diff
@@ -35,7 +35,9 @@ public void set(String mode) {
             String[] modes = StringUtils.split(mode, ',');
             world = AccessType.valueOf(modes[0].trim());
             trusted = AccessType.valueOf(modes[1].trim());
-        } catch (RuntimeException e) {}
+        } catch (RuntimeException e) {
+            LOGGER.error("RuntimeException ", e);
```
Comment (eshryane) [inline whois-commons/src/main/java/net/ripe/db/whois/common/MaintenanceMode.java]: How about LOGGER.error(e.getMessage(), e) ?
The exception class may not be RuntimeException

File: whois-commons/src/main/java/net/ripe/db/whois/common/dao/ResourceDataDao.java [hunk 5]
```diff
@@ -116,5 +116,10 @@ public int compareTo(final State other) {
                 }
             }
         }
+
+        @Override
```
Comment (eshryane) [inline whois-commons/src/main/java/net/ripe/db/whois/common/dao/ResourceDataDao.java]: This  is a useless method - it just calls the Object.equals() which is called anyway

File: whois-api/src/main/java/net/ripe/db/whois/api/transfer/logic/inetnum/InetnumTransfer.java (hunk 3 above)
Comment (petchells) [inline whois-api/src/main/java/net/ripe/db/whois/api/transfer/logic/inetnum/InetnumTransfer.java]: netname.getValue() already called on line 33, so the check isn't needed on l.34

File: whois-api/src/main/java/net/ripe/db/whois/api/autocomplete/AutocompleteService.java [hunk 6]
```diff
@@ -149,7 +139,7 @@ private Response badRequest(final String message) {
                     .entity(message).build();
     }
 
-    private Response ok(final Object message) {
+    private Response okRequest(final Object message) {
```
Comment (eshryane) [inline whois-api/src/main/java/net/ripe/db/whois/api/autocomplete/AutocompleteService.java]: This method is to create a response - maybe okResponse() is clearer?

File: whois-api/src/main/java/net/ripe/db/whois/api/autocomplete/AutocompleteService.java [hunk 7]
```diff
@@ -132,13 +122,13 @@ public Response lookup(
...
         return attributes.stream()
-                .map(attribute -> AttributeType.getByName(attribute))
+                .map(AttributeType::getByName)
                 .collect(Collectors.toSet());
     }
 
     private Set<ObjectType> getObjectTypes(final Collection<String> types) {
         return types.stream()
-                .map(type -> ObjectType.getByName(type))
+                .map(ObjectType::getByName)
```
Comment (eshryane) [inline whois-api/src/main/java/net/ripe/db/whois/api/autocomplete/AutocompleteService.java]: nice improvement

File: whois-api/src/main/java/net/ripe/db/whois/api/rest/DomainObjectService.java [hunk 8]
```diff
@@ -9,18 +9,7 @@
...
-import net.ripe.db.whois.update.domain.Operation;
-import net.ripe.db.whois.update.domain.Origin;
-import net.ripe.db.whois.update.domain.Paragraph;
-import net.ripe.db.whois.update.domain.PasswordCredential;
-import net.ripe.db.whois.update.domain.SsoCredential;
-import net.ripe.db.whois.update.domain.Update;
-import net.ripe.db.whois.update.domain.UpdateContext;
-import net.ripe.db.whois.update.domain.UpdateMessages;
-import net.ripe.db.whois.update.domain.UpdateStatus;
+import net.ripe.db.whois.update.domain.*;
```
Comment (eshryane) [inline whois-api/src/main/java/net/ripe/db/whois/api/rest/DomainObjectService.java]: We should leave existing imports as-is, don't replace with a *

File: whois-api/src/main/java/net/ripe/db/whois/api/rest/GeolocationService.java [hunk 9]
```diff
@@ -116,7 +117,7 @@ public WhoisResources geolocation(
                     }
                 }
             } catch (EmptyResultDataAccessException ignored) {
-                // stop looking
+                LOGGER.debug("EmptyResultDataAccessException {} in {} class ", ignored.getMessage(), ignored.getClass().getName());
```
Comment (eshryane) [inline whois-api/src/main/java/net/ripe/db/whois/api/rest/GeolocationService.java]: This exception is expected, and is thrown by the lookup() method, if the RPSL object no longer exists. Do we need to log it at all ?

File: whois-api/src/main/java/net/ripe/db/whois/api/rest/ReferencedTypeResolver.java [hunk 10]
```diff
@@ -81,6 +84,7 @@ public String getReferencedType(final AttributeType attributeType, final CIStrin
                                 // TODO: [AH] for each person or role reference returned, we make an sql lookup - baaad
                                 return rpslObjectDao.findByKey(objectType, value.toString()).getObjectType().getName();
                             } catch (EmptyResultDataAccessException ignored) {
+                                LOGGER.debug("ReferencedTypeResolver {} in {} class", ignored.getMessage(), ignored.getClass().getName());
```
Comment (eshryane) [inline whois-api/src/main/java/net/ripe/db/whois/api/rest/ReferencedTypeResolver.java]: "ReferencedTypeResolver" is already printed as the class name at the beginning of the Logger message, no need to add it again

File: whois-api/src/main/java/net/ripe/db/whois/api/rest/ReferencesService.java [hunk 11]
```diff
@@ -637,11 +579,15 @@ private RpslObjectWithReplacements replaceReferences(final RpslObject object, fi
         final Map<RpslAttribute, RpslAttribute> replacements = Maps.newHashMap();
 
         for (final RpslAttribute rpslAttribute : object.getAttributes()) {
-            for (final RpslObject reference : references) {
-                if (rpslAttribute.getCleanValue().equals(reference.getKey()) &&
-                        rpslAttribute.getType().getReferences().contains(ObjectType.PERSON) ||
-                        rpslAttribute.getType().getReferences().contains(ObjectType.ROLE)) {
-                    replacements.put(rpslAttribute, new RpslAttribute(rpslAttribute.getType(), dummyRole));
+            AttributeType rpsAttributeType = rpslAttribute.getType();
```
Comment (eshryane) [inline whois-api/src/main/java/net/ripe/db/whois/api/rest/ReferencesService.java]: use final keyword

File: whois-client/src/main/java/net/ripe/db/whois/api/rest/mapper/AttributeMapper.java [hunk 12]
```diff
@@ -2,10 +2,24 @@
...
 
 import java.util.Collection;
+import java.util.Collections;
 
 public interface AttributeMapper {
-    Collection<RpslAttribute> map(Attribute attributes);
+
     Collection<Attribute> map(RpslAttribute rpslObject, String source);
+
+    default Collection<RpslAttribute> map(Attribute attribute) {
```
Comment (eshryane) [inline whois-client/src/main/java/net/ripe/db/whois/api/rest/mapper/AttributeMapper.java]: good improvement!

File: whois-commons/src/main/java/net/ripe/db/whois/common/dao/ResourceDataDao.java [hunk 13]
```diff
@@ -116,5 +116,25 @@ public int compareTo(final State other) {
...
+        public boolean equals(Object o) {
+            if (this == o) return true;
+            if (o == null || getClass() != o.getClass()) return false;
+
+            State state = (State) o;
+
+            if (id != state.id) return false;
```
Comment (eshryane) [inline whois-commons/src/main/java/net/ripe/db/whois/common/dao/ResourceDataDao.java]: these three statements can be folded into one: return (id == state.id) && (count == state.count) && ...

File: whois-commons/src/main/java/net/ripe/db/whois/common/dao/jdbc/JdbcRpslObjectDao.java [hunk 14]
```diff
@@ -207,6 +207,7 @@ public RpslObject getByKeyOrNull(final ObjectType type, final CIString key) {
             try {
                 result.add(getByKey(type, searchKey));
             } catch (EmptyResultDataAccessException ignored) {
+                LOGGER.debug("EmptyResultDataAccessException {} in {} class ", ignored.getMessage(), ignored.getClass().getName());
```
Comment (eshryane) [inline whois-commons/src/main/java/net/ripe/db/whois/common/dao/jdbc/JdbcRpslObjectDao.java]: This is an expected exception, if there is no matching row in the database, can we ignore it instead?

File: whois-commons/src/main/java/net/ripe/db/whois/common/jdbc/driver/LoggingDriver.java [hunk 15]
```diff
@@ -50,6 +50,7 @@ public synchronized void init() {
                 throw new IllegalStateException("Unable to de-register logging JDBC driver", e);
             }
         } catch (SQLException expected) {
+            LOGGER.error("SQLException ", expected);
```
Comment (eshryane) [inline whois-commons/src/main/java/net/ripe/db/whois/common/jdbc/driver/LoggingDriver.java]: log @ debug for expected exception, or ignore

File: whois-query/src/main/java/net/ripe/db/whois/query/query/SearchKey.java [hunk 16]
```diff
@@ -6,10 +6,13 @@
 import net.ripe.db.whois.common.iptree.Ipv6RouteEntry;
 import net.ripe.db.whois.common.rpsl.attrs.AsBlockRange;
 import net.ripe.db.whois.common.rpsl.attrs.AttributeParseException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
 
 import java.util.regex.Pattern;
 
 class SearchKey {
+    private static final Logger LOGGER = LoggerFactory.getLogger(SearchKey.class);
```
Comment (eshryane) [inline whois-query/src/main/java/net/ripe/db/whois/query/query/SearchKey.java]: These error() statements are going to generate a lot of exceptions on bad user input, can they be ignored or log at debug

**Pull request merged**